tchannel/tornado/peer.py

```diff
@@ -618,7 +617,8 @@ def send_with_retry(self, request, peer, retry_limit, connection):
             try:
                 response = yield self._send(connection, request)
                 raise gen.Return(response)
-            except (ProtocolError, TimeoutError) as error:
+            # Why are we retying on all errors????
+            except TChannelError as error:
```

Comment:
There will be a problem here. Existing code use the ``code`` field in the ProtocolError to know if it is retry or not. If the new error type doesn't have ``code`` field, it will break.